lib/uring/uring_stubs.c

```diff
@@ -524,6 +497,136 @@ ocaml_uring_submit_statx_byte(value* values, int argc) {
 			  values[6]);
 }
 
+// Non-allocating (for native mode) accessors for statx
+#define STATX_GETTER(field, return_type, ocaml_value_maker) \
+return_type ocaml_uring_statx_##field##_native(value v_statx) { \
+  struct statx *s = Statx_val(v_statx); \
+  return s->stx_##field; \
+} \
+value ocaml_uring_statx_ ## field ## _bytes(value v_statx) { \
+  return ocaml_value_maker(ocaml_uring_statx_##field##_native(v_statx)); \
+}
+
+// Int64
+STATX_GETTER(blksize, int64_t, caml_copy_int64);
+STATX_GETTER(attributes, int64_t, caml_copy_int64);
+STATX_GETTER(nlink, int64_t, caml_copy_int64);
+STATX_GETTER(uid, int64_t, caml_copy_int64);
+STATX_GETTER(gid, int64_t, caml_copy_int64);
+STATX_GETTER(ino, int64_t, caml_copy_int64);
+STATX_GETTER(blocks, int64_t, caml_copy_int64);
+STATX_GETTER(attributes_mask, int64_t, caml_copy_int64);
+STATX_GETTER(mask, int64_t, caml_copy_int64);
+
```

Comment:
Minor: it would be easier to read with the field name at the end, so the bits that don't change line up vertically:
```suggestion
STATX_GETTER(int64_t, caml_copy_int64, blksize);
STATX_GETTER(int64_t, caml_copy_int64, attributes);
STATX_GETTER(int64_t, caml_copy_int64, nlink);
STATX_GETTER(int64_t, caml_copy_int64, uid);
STATX_GETTER(int64_t, caml_copy_int64, gid);
STATX_GETTER(int64_t, caml_copy_int64, ino);
STATX_GETTER(int64_t, caml_copy_int64, blocks);
STATX_GETTER(int64_t, caml_copy_int64, attributes_mask);
STATX_GETTER(int64_t, caml_copy_int64, mask);
```